### PR TITLE
Fix style for STS by group

### PR DESCRIPTION
**Pull request by flomnes**
- Remove magic values (0, 1, 2), use `enum`
- Add a few `const` qualifiers
- Add const reference to results to improve readibility
- Formatting

### CODE REVIEW

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h [hunk 1]
```diff
@@ -359,30 +378,25 @@ class STSbyGroup : public Variable::IVariable<STSbyGroup<NextT>,
...
 
-
 private:
     //! Intermediate values for each year
     typename VCardType::IntermediateValuesType pValuesForTheCurrentYear;
     size_t nbColumns_ = 0;
     std::vector<std::string> groupNames_; // Names of group containing the clusters of the area
     std::map<std::string, unsigned int> groupToNumbers_; // Gives to each group (of area) a number
-    const std::vector<std::string> VAR_POSSIBLE_KINDS = { "injection", "withdrawal", "level" };
+    const std::vector<std::string> VAR_POSSIBLE_KINDS = {"injection", "withdrawal", "level"};
```
Comment (guilpier-code) [inline src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h]: You didn't move **VAR_POSSIBLE_KINDS** to **caption(...)** as you said you would ?

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h [hunk 2]
```diff
@@ -88,18 +88,23 @@ struct VCardSTSbyGroup
 ** \brief Variables related to short term storage groups
 */
 template<class NextT = Container::EndOfList>
-class STSbyGroup : public Variable::IVariable<STSbyGroup<NextT>,
-                                              NextT,
-                                              VCardSTSbyGroup>
+class STSbyGroup : public Variable::IVariable<STSbyGroup<NextT>, NextT, VCardSTSbyGroup>
 {
+private:
```
Comment (guilpier-code) [inline src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: There is already a **private** section. **VariableType** should be moved down there.
Furthermore, why not 
```C++
enum class VariableType { injection, withdrawal, level };
```
instead ?

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h (hunk 2 above)
Comment (flomnes) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: Yes, but then we must make it convertible to int

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h (hunk 1 above)
Comment (flomnes) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h]: No, I leave up to you

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h (hunk 2 above)
Comment (guilpier-code) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: ok, then would 
```C++
enum VariableType { injection = 0, withdrawal, level };
```
suit ?

Comment (flomnes) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: More like
```cpp
enum class VariableType : std::size_t { injection = 0, withdrawal = 1, level = 2};
```

Comment (guilpier-code) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: ok, it seems good to me

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h (hunk 1 above)
Comment (flomnes) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h]: done

File: src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h (hunk 2 above)
Comment (flomnes) [reply src/solver/variable/include/antares/solver/variable/economy/STSbyGroup.h:93]: We would need to `static_cast` values, which we don't really want

**Pull request merged**